File: backend/app/schemas/sync.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SyncLogResponse(BaseModel):
    id: int
    sync_type: str
    status: str
    records_processed: Optional[int] = None
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class SyncStatusResponse(BaseModel):
    is_running: bool
    latest_sync: Optional[SyncLogResponse] = None

File: frontend/src/services/sync.ts
import api from './api';

export interface SyncLog {
  id: number;
  sync_type: string;
  status: string;
  records_processed: number | null;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
}

export interface SyncStatus {
  is_running: boolean;
  latest_sync: SyncLog | null;
}

export async function triggerLDAPSync(): Promise<SyncLog> {
  const { data } = await api.post('/sync/ldap');
  return data;
}

export async function triggerUserStatusSync(): Promise<SyncLog> {
  const { data } = await api.post('/sync/ldap/user-status');
  return data;
}

export async function getSyncLogs(params?: Record<string, unknown>) {
  const { data } = await api.get('/sync/logs', { params });
  return data;
}

export async function getSyncStatus(): Promise<SyncStatus> {
  const { data } = await api.get('/sync/status');
  return data;
}
